cas-client-core/src/main/java/org/jasig/cas/client/util/CommonUtils.java

```diff
@@ -262,12 +264,44 @@ private static boolean requestIsOnStandardPort(final HttpServletRequest request)
     /**
      * Constructs a service url from the HttpServletRequest or from the given
      * serviceUrl. Prefers the serviceUrl provided if both a serviceUrl and a
+     * serviceName. Compiles a list of all service parameters for supported protocols
+     * and removes them all from the query string.
+     *
+     * @param request the HttpServletRequest
+     * @param response the HttpServletResponse
+     * @param service the configured service url (this will be used if not null)
+     * @param serverNames the server name to  use to construct the service url if the service param is empty.  Note, prior to CAS Client 3.3, this was a single value.
+     *           As of 3.3, it can be a space-separated value.  We keep it as a single value, but will convert it to an array internally to get the matching value. This keeps backward compatability with anything using this public
+     *           method.
+     * @param artifactParameterName the artifact parameter name to remove (i.e. ticket)
+     * @param encode whether to encode the url or not (i.e. Jsession).
+     * @return the service url to use.
+     */
+    @Deprecated
+    public static String constructServiceUrl(final HttpServletRequest request, final HttpServletResponse response,
+                                             final String service, final String serverNames,
+                                             final String artifactParameterName, final boolean encode) {
+        final Set<String> serviceParameterSet = new HashSet<String>(4);
+        for (final Protocol protocol : Protocol.values()) {
+            serviceParameterSet.add(protocol.getServiceParameterName());
+        }
+        final String serviceParameterNames = serviceParameterSet.toString()
```

Comment:
Because of this issue: https://github.com/Jasig/java-cas-client/issues/104

We should make sure all protocol-specific parameters are removed from the service url. This will work for both service and TARGET. Though, I understand that for BC reasons we could just remove "service" but that would sort of be "buggy" if someone actually calls that method with "TARGET" in the URL for example.